### PR TITLE
Fix WinHttp SSL validation and other small fixes

### PR DESCRIPTION
**Pull request by jgalvan**
- When SSL validation was added, it was only enabled for specific platforms, however there was one piece of code that was left without the platform check. This would break any other platform that doesn't use it. Original PR for reference: #517
- Fixing C4701/C4703 for uninitialized variables

### CODE REVIEW

File: Source/HTTP/WinHttp/winhttp_http_task.cpp [hunk 1]
```diff
@@ -1208,6 +1208,7 @@ HRESULT winhttp_http_task::send(
         return HRESULT_FROM_WIN32(dwError);
     }
 
+#if HC_PLATFORM == HC_PLATFORM_WIN32 || HC_PLATFORM == HC_PLATFORM_GDK
```
Comment (lubeltra) [inline Source/HTTP/WinHttp/winhttp_http_task.cpp:1211]: As I mentioned offline, I think this code should be on for GRTS

**Review by lubeltra — CHANGES_REQUESTED** [review]

File: Source/HTTP/WinHttp/winhttp_http_task.cpp (hunk 1 above)
Comment (sashaweiss-msft) [reply Source/HTTP/WinHttp/winhttp_http_task.cpp:1211]: but there's no GRTS `HC_PLATFORM`, right?

**Review by sashaweiss-msft — APPROVED** [review]

**Review by lubeltra — APPROVED** [review]

**Review by jasonsandlin — APPROVED** [review]

**Comment by boczar** [comment]
Hi folks,

Sorry for digging this up two months later, but how is this supposed to work?

#if HC_PLATFORM_IS_MICROSOFT && !HC_PLATFORM_UWP && !HC_PLAFTORM_XDK

HC_PLATFORM_UWP is defined as '2'. Always. So the above statement will always be false, regardless what platform you build on right? Or am I missing something?

Previous version was functionally correct:
#if HC_PLATFORM == HC_PLATFORM_WIN32 || HC_PLATFORM == HC_PLATFORM_GDK
Although it could be adapted to match the change like this:
#if HC_PLATFORM_IS_MICROSOFT && (HC_PLATFORM != HC_PLATFORM_UWP) && (HC_PLATFORM != HC_PLAFTORM_XDK)

**Comment by jplafonta** [comment]
I agree, this looks incorrect.  I will PR a fix for this and do a quick scan for other spots where it might have been messed up

---
In reply to: [990140683](https://github.com/microsoft/libHttpClient/pull/620#issuecomment-990140683) [](http://example.com/codeflow?ancestors=990140683,990140683)

**Comment by jplafonta** [comment]
@boczar Looks like this has since been fixed.

---
In reply to: [990227793](https://github.com/microsoft/libHttpClient/pull/620#issuecomment-990227793) [](http://example.com/codeflow?ancestors=990227793,990140683)

**Pull request merged**